models/articles.models.js

```diff
@@ -40,14 +41,49 @@ exports.checkArticleExists = async (article_id) => {
     return Promise.reject({ status: 404, message: 'invalid article id' });
   } else return results.rows[0];
 };
-exports.selectAllArticles = async () => {
-  const results = await db.query(
-    `SELECT articles.*, COUNT(comment_id) AS comment_count
-    FROM articles
-    JOIN comments
-    ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;`
-  );
+exports.selectAllArticles = async (
+  sort_by = 'created_at',
+  order = 'desc',
+  topic
+) => {
+  const topics = await selectAllTopics();
+  topicsArray = [];
+  topics.forEach((object) => {
+    topicsArray.push(object.slug);
+  });
```

Comment:
This is a longwinded way of getting an array of topic slugs. Think of a better array method you could use here. I bet you can get this down to one line!